pandas/tests/tools/test_concat.py

```diff
@@ -1048,6 +1048,24 @@ def test_concat_multiindex_with_tz(self):
         result = concat([df, df])
         tm.assert_frame_equal(result, expected)
 
+    def test_concat_multiindex_with_none_in_index_names(self):
+        # GH 15787
+        from pandas.indexes.frozen import FrozenList
+
+        index = pd.MultiIndex.from_product([[1], range(5)],
+                                           names=['level1', None])
+        df = pd.DataFrame({'col': range(5)}, index=index)
+
+        result = concat([df, df], keys=[1, 2], names=['level2'])
+        result = result.index.names
```

Comment:
I would rather you actually construct the expected DF here (and use ``assert_frame_equal``